### PR TITLE
File property feature

### PR DESCRIPTION
**Pull request by VirajHasitha**
## Purpose
This feature supports read properties from a file and enable parameterizing the synapse configurations and key-value will be resolved using the file property resolver when deploying a new artifact.  
This feature will use the org.apache.synapse.commons.resolvers.ResolverFactory whenever instantiate.

## Test environment
micro-integrator

### CODE REVIEW

**Comment by VirajHasitha** [comment]
This PR should merge after merging the synapse PR https://github.com/wso2/wso2-synapse/pull/1431

File: distribution/src/conf/file.properties [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+stockQuoteEP=http://localhost:9000/services/SimpleStockQuoteService
```
Comment (Nirothipan) [inline distribution/src/conf/file.properties]: Shall we add a new line at the EOF

Comment (VirajHasitha) [reply distribution/src/conf/file.properties]: Fix with the commit https://github.com/wso2/micro-integrator/commit/ccb931501ef0434ae0955d1b64e04d8c5845360b

File: integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/synapse/common/resolvers/FilePropertyResolverTestCase.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package org.wso2.carbon.esb.synapse.common.resolvers;
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/synapse/common/resolvers/FilePropertyResolverTestCase.java]: Let's add the license header

**Pull request merged**